pyblish_bumpybox/plugins/collect_existing_files.py

```diff
@@ -27,9 +30,52 @@ def version_get(self, string, prefix):
         matches = re.findall(regex, string, re.IGNORECASE)
         if not len(matches):
             msg = "No \"_"+prefix+"#\" found in \""+string+"\""
-            raise ValueError(msg)
+            self.log.debug(msg)
+            return None
         return matches[-1:][0][1], re.search("\d+", matches[-1:][0]).group()
 
+    def scan_versions(self, instance_collection, version):
+
+        # Getting collections of all previous versions and current version
+        collections = []
+        for count in range(1, int(version) + 1):
+
+            # Generate collection
+            version_string = "v" + str(count).zfill(len(version))
+            head = instance_collection.head.replace(
+                "v" + version, version_string
+            )
+            collection = clique.Collection(
+                head=head.replace("\\", "/"),
+                padding=instance_collection.padding,
+                tail=instance_collection.tail
+            )
+            collection.version = count
+
+            collection = self.scan_collection(collection)
+
+            if list(collection):
+                collections.append(collection)
+
+        return collections
+
+    def scan_collection(self, collection):
+
+        # Scan collection directory
+        scan_dir = os.path.dirname(collection.head)
+        if scan_dir not in self.scanned_dirs and os.path.exists(scan_dir):
+            for f in os.listdir(scan_dir):
+                file_path = os.path.join(scan_dir, f)
+                self.files.append(file_path.replace("\\", "/"))
+            self.scanned_dirs.append(scan_dir)
```

Comment:
Sure, writing smaller functions that do less is generally encouraged given it offers greater flexibility on how those functions might be used and cleaner simpler code.

Generally you write functions that perform a single thing - which is simpler to test. With smaller functions, you can write tests for each smaller function with simpler logic. What this function is doing is really a scan_and_match, so any test you might write has to test both behaviours. That can lead to confusing tests that are trying to test more complex logic. 

Much of this is hypothetical given there are currently no unit tests for this repo.